### PR TITLE
[TD]Leader and Welding Symbol (fix #18094)

### LINKED ISSUE #18094: TechDraw: Leader and Welding Symbol Problems on Rotated Base View
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Problem reported on forum: https://forum.freecad.org/viewtopic.php?t=92072

1) auto horizontal option corrupts the last two waypoints, making them identical and thereby removing the last segment of the leader line.  The last two waypoints are used to determine the position of the kink point and weld symbols

2) weld symbol is not aware of the rotation of the parent leader's base view.  this causes the weld symbol to be positioned in the unrotated position and far from the leader line.

### Full version info

```shell
OS: Linux Mint 21.2 (X-Cinnamon/cinnamon/xcb)
Architecture: x86_64
Version: 1.1.0dev.39189 +1 (Git)
Build type: debug
Branch: LeaderWelding
Hash: 4a9d20d2f626a5326911647d367ab6716c653acb
Python 3.10.12, Qt 5.15.2, Coin 4.0.0, Vtk , OCC 7.8.0
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * fasteners 0.5.21
```

### Subproject(s) affected?

None

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by WandererFan**
This PR implements a fix for issue #18094.

LeaderWeldingProblem_3Points_variousDirs

### CODE REVIEW

File: src/Mod/TechDraw/Gui/QGILeaderLine.cpp [hunk 1]
```diff
@@ -504,20 +498,19 @@ std::vector<QPointF> QGILeaderLine::getWayPointsFromFeature()
...
-            qPoints.push_back(DU::toQPointF(DGU::toSceneCoords(entry, false)));
+            qPoints.push_back(DU::toQPointF(Rez::guiX(entry)));
         }
     }
 
     if (qPoints.empty()) {
         Base::Console().Warning("QGILeaderLine::getWayPointsFromFeature - no points\n");
     }
-
+
```
Comment (chennes) [inline src/Mod/TechDraw/Gui/QGILeaderLine.cpp:508]:
```suggestion

```

**Comment by chennes** [comment]
We are going to drop the `[readability-identifier-length` linter check, so we are ignoring that one here.

**Comment by chennes** [comment]
Pinging myself to do this review... @chennes

**Comment by maxwxyz** [comment]
pinging @chennes to remind him of his self-ping 😄

**Comment by adrianinsaval** [comment]
should this be backported @WandererFan ?

**Pull request merged**